test/test_framework.py

```diff
@@ -1397,10 +1456,13 @@ def save_snapshot(self, value: typing.Union[ops.StoredStateData, ops.EventBase])
                     self.snapshots.append((type(value), value.snapshot()))
                 return super().save_snapshot(value)
 
+        tmpdir = Path(tempfile.mkdtemp())
```

Comment:
Can we not use the pytest `tmpdir` fixture for most of these?

Reply:
This isn't a fixture, it's creating a tmp dir manually using tempfile. The name is confusing because [pytest does have a fixture named `tmpdir` (although it's preferred to use `tmp_path`).](https://docs.pytest.org/en/7.1.x/reference/reference.html#tmpdir)